core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -43,6 +52,7 @@
  * This table may return duplicate rows.
  */
 public class AllManifestsTable extends BaseMetadataTable {
+  private static final int REF_SNAPSHOT_ID = 18;
```

Comment:
Side comment / non-blocking: I'm a _big_ fan of using named constants for schema element IDs.

Should we update to use named position constants throughout, particularly for schemas like the metadata tables that are evolving (note that `content` is listed first but has ID 14). But is the cherry-picking that would be required by a large diff like this considered an unnecessary change?

Reply:
Yea I am not sure if there is consensus, Russell made a discussion in https://github.com/apache/iceberg/pull/4847#discussion_r906391975 but didn't finished (not sure if there is a big value for the other constants if they are never referenced outside?).  But If so, we should do it in another pr as its unrelated.